src/ebmc/property_checker.cpp

```diff
@@ -356,6 +356,56 @@ property_checker_resultt bit_level_bmc(
   }
 }
 
+property_checker_resultt engine_heuristic(
+  const cmdlinet &cmdline,
+  transition_systemt &transition_system,
+  ebmc_propertiest &properties,
+  message_handlert &message_handler)
+{
+  messaget message(message_handler);
+
+  if(properties.properties.empty())
+  {
+    message.error() << "no properties" << messaget::eom;
+    return property_checker_resultt::error();
+  }
+
+  auto solver_factory = ebmc_solver_factory(cmdline);
+
+  if(!properties.has_unknown_property())
+    return property_checker_resultt{properties}; // done
+
+  // First try 1-induction, word-level
+  message.status() << "Attempting 1-induction" << messaget::eom;
```

Comment:
How about inserting one more line of output that tells the user that they haven't specified an engine and thereby triggered heuristic (maybe it should actually be called "default"?) selection?